xcube/core/subsampling.py

```diff
@@ -0,0 +1,107 @@
+#  The MIT License (MIT)
+#  Copyright (c) 2022 by the xcube development team and contributors
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a
+#  copy of this software and associated documentation files (the "Software"),
+#  to deal in the Software without restriction, including without limitation
+#  the rights to use, copy, modify, merge, publish, distribute, sublicense,
+#  and/or sell copies of the Software, and to permit persons to whom the
+#  Software is furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included in
+#  all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+#  DEALINGS IN THE SOFTWARE.
+
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a
+#  copy of this software and associated documentation files (the "Software"),
+#  to deal in the Software without restriction, including without limitation
+#  the rights to use, copy, modify, merge, publish, distribute, sublicense,
+#  and/or sell copies of the Software, and to permit persons to whom the
+#  Software is furnished to do so, subject to the following conditions:
+#
+#
+from typing import Dict, Tuple, Hashable, Optional
+
+import xarray as xr
+
+from xcube.util.assertions import assert_instance
+
+
+def subsample_dataset(
+        dataset: xr.Dataset,
+        step: int,
+        xy_dim_names: Optional[Tuple[str, str]] = None,
+) -> xr.Dataset:
+    """
+    Subsample *dataset* with given integer subsampling *step*.
+    Only data variables with spatial dimensions given by
+    *xy_dim_names* are subsampled.
+
+    :param dataset: the dataset providing the variables
+    :param step: the integer subsampling step
+    :param xy_dim_names: the spatial dimension names
+    """
+    assert_instance(dataset, xr.Dataset, name='dataset')
+    assert_instance(step, int, name='step')
+    var_slices = get_dataset_subsampling_slices(dataset,
+                                                step,
+                                                xy_dim_names)
+    if not var_slices:
+        return dataset
+    return xr.Dataset(
+        data_vars={
+            var_name: (var[var_slices[var_name]]
+                       if var_name in var_slices else var)
+            for var_name, var in dataset.data_vars.items()
+        },
+        attrs=dataset.attrs
+    )
+
+
+_EMPTY_SLICE = slice(None, None, None)
```

Comment:
I see that this name makes sense, but for understanding what is the purpose of this slice, the term _FULL_SLICE would be better. I'd definitely prefer it.